packages/freezed/lib/src/templates/tear_off.dart

```diff
@@ -19,6 +19,11 @@ class TearOff {
   final GenericsDefinitionTemplate genericsDefinition;
   final List<ConstructorDetails> allConstructors;
 
+  bool get _hasGenericArgumentFactories =>
+      allConstructors.any((cons) => cons.decorators
+          .where((dec) => dec.startsWith('@JsonSerializable'))
+          .any((dec) => dec.contains('genericArgumentFactories: true')));
```

Comment:
We can't rely on the "toString" of the annotation

It should be possible to do:

```dart
const alias = JsonSerializable(genericArgumentFactories: true);

...
@alias
factory MyClass() = _MyClass;
```

Reply:
Yep, I was thinking on doing something like this:

```dart
  bool get hasGenericArgumentFactories {
    if (!hasJsonSerializable) return false;
    final annotation =
        const TypeChecker.fromRuntime(JsonSerializable).firstAnnotationOf(this);
    return annotation.getField('genericArgumentFactories').toBoolValue();
  }
```

But I have not been able to add a field to the `ConstructorDetails` class